packages/platform/atoms/hooks/connect/useConnect.ts

```diff
@@ -44,3 +44,35 @@ export const useConnect = (calendar: (typeof CALENDARS)[number], redir?: string)
 
   return { connect };
 };
+
+export const useSaveCalendarCredentials = (
+  calendar: (typeof CALENDARS)[number],
+  username: string,
+  password: string
+) => {
+  const body = {
+    username,
+    password,
+  };
+
+  const status = useQuery({
+    queryKey: getQueryKey(calendar),
+    staleTime: Infinity,
+    enabled: false,
+    queryFn: () => {
+      return http
+        ?.post<ApiResponse<{ status: string }>>(`/calendars/${calendar}/save}`, {
+          body,
+        })
+        .then(({ data: responseBody }) => {
+          if (responseBody.status === SUCCESS_STATUS) {
+            return responseBody.data.status;
+          }
+          if (responseBody.status === ERROR_STATUS) throw new Error(responseBody.error.message);
+          return "";
+        });
+    },
+  });
```

Comment:
this should be a useMutation not a useQuery, useMutation does not need queryKey, mutation should be called when save button is clicked, if there is an error, keep the dialog open and display error, if its success, close the dialog and force useCheck to run again